### DISCUSSION TITLE
How to encode more complex data types like structs and bytes?

### DISCUSSION THREAD

**Discussion by zkSoju** [wevm/wagmi#531] [General]
I'm trying to use `useContractRead` with this function `quoteLayerZeroFee(uint16 chainId, bytes memory payload, IStargateRouter.lzTxObj memory lzParams)` but it's giving me this error for all three inputs, which makes me think that it's trying to convert each argument to a BigNumber: 

```Error: invalid BigNumber value (argument="value", value=undefined, code=INVALID_ARGUMENT, version=bignumber/5.6.2)```

I'm using `encode` from ethers to get the bytes and an array to create the struct.

**Comment by tmm** [comment]
`useContractRead` doesn't force anything internally. If you post a minimal example here, maybe someone can take a look.

**Reply by zkSoju** [reply]
Oh I see. Sure!

Smart contract
```solidity
   function quoteLayerZeroFee(
        uint16 chainId,
        bytes memory payload,
        IStargateRouter.lzTxObj memory lzParams
    )
...
    struct lzTxObj {
        uint256 dstGasForCall;
        uint256 dstNativeAmount;
        bytes dstNativeAddr;
    }
```

Frontend
 ```javascript
const { data: feeData } = useContractRead(
    {
      addressOrName: contractAddress,
      contractInterface: ABI,
    },
    "quoteLayerZeroFee",
    {
      args: [
        10010,
        abiCoder.encode(
          ["address", "uint", "address"],
          [
            "0x79092A805f1cf9B0F5bE3c5A296De6e51c1DEd54",
            0,
            "0x79092A805f1cf9B0F5bE3c5A296De6e51c1DEd54",
          ]
        ),
        [500000, 0, "0x"],
      ],
    }
  );
```

I also tried some versions of encode for the struct as well but with no luck.

**Comment by zkSoju** [comment]
Hi still unable to resolve this if anyone has any ideas thanks!
